### ISSUE TITLE
Request with POST or PATCH doesn't work

### ISSUE THREAD

**Issue by ghunti** [twg/devour#42]
Hi.

When you try to perform an arbitrary request with methods `PATCH` or `POST`, you end up getting an error: `TypeError: Cannot read property 'options' of undefined`.
I've tracked this down to the `_serialize.js` file, method `resource`. The problem is that in line 11 `let model = this.modelFor(modelName)` since we are not providing a model, the model variable is undefined and when we call `model.options` latter, the error is triggered.

Ex of calling code: `jsonApi.request('https://example.com', 'POST', { a_query_param: 3 }, { some_payload_item: 'blah' })`

Thanks

**Comment by Emerson** [comment]
Hmmmm – yea that would happen. How would you serialize this payload into a JSON API compliant request?

**Comment by ghunti** [comment]
Hi @Emerson 
In my personal case it would work, because I was doing the login and register, and they both return "users" type, respecting the JSON api format.

**Comment by Nainterceptor** [comment]
Hi @ghunti, I've encountered the same issue, right now.

Cause : 
=====
When you call request(), devour basically call every middleware in a promise.
Patch & Post middlewares try to serialize payload, and, in this serialize, devour try to fetch model. ([right here](https://github.com/twg/devour/blob/master/src/middleware/json-api/_serialize.js#L10)).
But, request don't know what's model of the data, so model = undefined.

Workaround 1 : 
==========
Most simple may be to add `|| {}` on this line, but it's the most dirty too, because this error is useful while development.

Workaround 2 (Probably the best one) : 
==========
Add another parameter on request to be able to override req object to inject model name

Workaround 3 :
===========
Don't call request(), but call runMiddleware(). I suggest a simpler function name like "rawRequest" for that, like : 

```javascript
 rawRequest(url, method = 'GET', req = {}) {
    Object.assign(req, { url, method })
    return this.runMiddleware(req)
  }
```

**Comment by Emerson** [comment]
I think this is now released in the 2.0 release!

**Comment by yaodev778** [comment]
Hello, Emerson
I am also facing this issue. I am using devour 2.0.0.
